Repositories/UserRepository.cs

```diff
@@ -185,7 +190,17 @@ public bool UserWithRoleExists(Role role)
                 .SingleOrDefault(r => r.Role.Id == role.Id) != null;
         }
 
+        public async Task<List<User>> GetAllExpectedGraduatingUsers()
+        {
+            DateTime now = DateTime.Now;
+            DateTime max = DateTime.Now.AddMonths(6);
```

Comment:
I don't think the decision about how many months are added to define expected graduating users should be in the repository. This is a business logic rule and should therefore in my opinion be in the service, passing it to the repository as a parameter.